From 37095546fc7eca29d5a790a7d7d0a1314b173dd3 Mon Sep 17 00:00:00 2001
From: Raef Coles <raef.coles@arm.com>
Date: Wed, 2 Sep 2020 10:04:05 +0100
Subject: [PATCH 3/6] Add include/mbedtls to include property of targets

Signed-off-by: Raef Coles <raef.coles@arm.com>
---
 CMakeLists.txt         | 3 +--
 library/CMakeLists.txt | 8 ++++++--
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d970b9fd..949a50b2b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -217,14 +217,13 @@ if(ENABLE_ZLIB_SUPPORT)
     endif(ZLIB_FOUND)
 endif(ENABLE_ZLIB_SUPPORT)
 
-add_subdirectory(include)
-
 add_subdirectory(3rdparty)
 include_directories(${thirdparty_inc})
 list(APPEND libs ${thirdparty_lib})
 add_definitions(${thirdparty_def})
 
 add_subdirectory(library)
+add_subdirectory(include)
 
 if(ENABLE_PROGRAMS)
     add_subdirectory(programs)
diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index fe74de25d..02cd157bb 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -192,7 +192,9 @@ if(USE_STATIC_MBEDTLS_LIBRARY)
     set_target_properties(${mbedcrypto_static_target} PROPERTIES OUTPUT_NAME mbedcrypto)
     target_link_libraries(${mbedcrypto_static_target} ${libs})
     target_include_directories(${mbedcrypto_static_target}
-        PUBLIC ${MBEDTLS_DIR}/include/)
+        PUBLIC ${MBEDTLS_DIR}/include/
+        PUBLIC ${MBEDTLS_DIR}/include/mbedtls/
+    )
 
     add_library(${mbedx509_static_target} STATIC ${src_x509})
     set_target_properties(${mbedx509_static_target} PROPERTIES OUTPUT_NAME mbedx509)
@@ -213,7 +215,9 @@ if(USE_SHARED_MBEDTLS_LIBRARY)
     set_target_properties(${mbedcrypto_target} PROPERTIES VERSION 2.23.0 SOVERSION 5)
     target_link_libraries(${mbedcrypto_target} ${libs})
     target_include_directories(${mbedcrypto_target}
-        PUBLIC ${MBEDTLS_DIR}/include/)
+        PUBLIC ${MBEDTLS_DIR}/include/
+        PUBLIC ${MBEDTLS_DIR}/include/mbedtls/
+    )
 
     add_library(${mbedx509_target} SHARED ${src_x509})
     set_target_properties(${mbedx509_target} PROPERTIES VERSION 2.23.0 SOVERSION 1)
-- 
2.20.1

