From 1c4af63fc32e16022f884e29ad3ccbdb8c3ed2f2 Mon Sep 17 00:00:00 2001
From: Raef Coles <raef.coles@arm.com>
Date: Wed, 2 Sep 2020 10:15:02 +0100
Subject: [PATCH 3/4] Remove overriding of CMAKE_BUILD_TYPE

Signed-off-by: Raef Coles <raef.coles@arm.com>
---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba9227ff3..a671575b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,10 +112,6 @@ option: \n\
     endif()
 endif()
 
-set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}
-    CACHE STRING "Choose the type of build: None Debug Release Coverage ASan ASanDbg MemSan MemSanDbg Check CheckFull"
-    FORCE)
-
 # Create a symbolic link from ${base_name} in the binary directory
 # to the corresponding path in the source directory.
 function(link_to_source base_name)
-- 
2.20.1

